### PR TITLE
:hammer: refactor optim

### PR DESCRIPTION
**Pull request by kartik4949**
fixes optim todos by @geohot

### CODE REVIEW

**Comment by geohot** [comment]
You can go further and go eps and lr and stuff too.

**Comment by kartik4949** [comment]
> You can go further and do eps and lr and stuff too.

sure if u say so, i was waiting on div :)

File: tinygrad/optim.py [hunk 1]
```diff
@@ -26,27 +23,27 @@ def step(self):
...
     for i, t in enumerate(self.params):
-      self.v[i] = self.decay * self.v[i] + (self.one - self.decay) * t.grad.pow(self.two)
+      self.v[i] = self.decay * self.v[i] + (1.0 - self.decay) * t.grad.pow(2.0)
       t -= self.lr.div(self.v[i].sqrt() + self.eps) * t.grad
 
 class Adam(Optimizer):
   def __init__(self, params, lr=0.001, b1=0.9, b2=0.999, eps=1e-8):
     super(Adam, self).__init__(params)
-    self.lr, self.b1, self.b2, self.eps, self.t, self.one, self.two = [self.num(x) for x in [lr, b1, b2, eps, 0, 1, 2]]
+    self.lr, self.b1, self.b2, self.eps, self.t = lr, b1, b2, eps, Tensor([0.0], gpu=self.params[0].gpu, requires_grad=False)
```
Comment (geohot) [inline tinygrad/optim.py:38]: Shouldn't have to be a tensor

File: tinygrad/optim.py [hunk 2]
```diff
@@ -26,27 +23,27 @@ def step(self):
...
-    self.lr, self.decay, self.eps, self.one, self.two = [self.num(x) for x in [lr, decay, eps, 1, 2]]
+    self.lr, self.decay, self.eps = Tensor([lr], gpu=self.params[0].gpu, requires_grad=False), decay, eps
 
     self.v = [Tensor(np.zeros(t.shape, dtype=np.float32), gpu=params[0].gpu, requires_grad=False) for t in self.params]
 
   def step(self):
     for i, t in enumerate(self.params):
-      self.v[i] = self.decay * self.v[i] + (self.one - self.decay) * t.grad.pow(self.two)
+      self.v[i] = self.decay * self.v[i] + (1.0 - self.decay) * t.grad.pow(2.0)
       t -= self.lr.div(self.v[i].sqrt() + self.eps) * t.grad
```
Comment (geohot) [inline tinygrad/optim.py:33]: (t.grad * self.lr).div(self.v[i].sqrt() + self.eps)

Comment (geohot) [inline tinygrad/optim.py:33]: (t.grad * self.lr).div(self.v[i].sqrt() + self.eps)

**Pull request merged**